tests/unit/utils/test_urlutils.py

```diff
@@ -531,23 +531,46 @@ class TestIncDecNumber:
 
     """Tests for urlutils.incdec_number()."""
 
-    @pytest.mark.parametrize('url, incdec, output', [
-        ("http://example.com/index1.html", "increment", "http://example.com/index2.html"),
-        ("http://foo.bar/folder_1/image_2", "increment", "http://foo.bar/folder_1/image_3"),
-        ("http://bbc.c0.uk:80/story_1", "increment", "http://bbc.c0.uk:80/story_2"),
-        ("http://mydomain.tld/1_%C3%A4", "increment", "http://mydomain.tld/2_%C3%A4"),
-        ("http://example.com/site/5#5", "increment", "http://example.com/site/6#5"),
-
-        ("http://example.com/index10.html", "decrement", "http://example.com/index9.html"),
-        ("http://foo.bar/folder_1/image_3", "decrement", "http://foo.bar/folder_1/image_2"),
-        ("http://bbc.c0.uk:80/story_1", "decrement", "http://bbc.c0.uk:80/story_0"),
-        ("http://mydomain.tld/2_%C3%A4", "decrement", "http://mydomain.tld/1_%C3%A4"),
-        ("http://example.com/site/5#5", "decrement", "http://example.com/site/4#5"),
+    @pytest.mark.parametrize('incdec', ['increment', 'decrement'])
+    @pytest.mark.parametrize('value', [
+        '{}foo', 'foo{}', 'foo{}bar', '42foo{}'
     ])
-    def test_incdec_number(self, url, incdec, output):
+    @pytest.mark.parametrize('url', [
+        'http://example.com:80/v1/path/{}/test',
+        'http://example.com:80/v1/query_test?value={}',
+        'http://example.com:80/v1/anchor_test#{}',
+        'http://host_{}_test.com:80',
+        'http://m4ny.c0m:80/number5/3very?where=yes#{}'
+    ])
+    def test_incdec_number(self, incdec, value, url):
         """Test incdec_number with valid URLs."""
-        new_url = urlutils.incdec_number(QUrl(url), incdec)
-        assert new_url == QUrl(output)
+        # The integer used should not affect test output, as long as it's
+        # bigger than 1
+        # 4 was chosen by dice roll, guaranteed to be random
+        base_value = value.format(4)
```

Comment:
I think it'd be better to use a two-digit number, say `23` - otherwise I think we're in the situation again where the regex could do `\d` instead of `\d+` or whatever and the tests wouldn't notice.